yesod-test/Yesod/Test.hs

```diff
@@ -958,6 +959,40 @@ genericNameFromHTML match label html =
         name:_ -> Right name
     _ -> Left $ "More than one label contained " <> label
 
+
+-- | Choose a given value from a radio input, identified by the value of it's corresponding label.
+-- This is a very simple and naive function, which could fail in more complex scenarios. Potentially still quite useful.
+--
+-- Given this HTML, and we want to submit @color=Red@ to the server:
+--
+-- > <form method="POST">
+-- >   <label for="color-red">Red</label>
+-- >   <input id="color-red" name="color" value="Red" />
+-- >   <label for="color-blue">Blue</label>
+-- >   <input id="color-blue" name="color" value="Blue" />
+-- > </form>
+--
+-- You can set this parameter like so:
+--
+-- > request $ do
+-- >   chooseRadio "Red"
+--
+-- @since 1.6.17
+chooseRadio :: T.Text -- ^ The text contained within the @\<label>@.
+       -> RequestBuilder site ()
+chooseRadio v = do
+  mres <- fmap rbdResponse getSIO
+  res <-
+    case mres of
+      Nothing -> failure "chooseRadio: No response available"
+      Just res -> return res
+  let body = simpleBody res
+
+  let name = genericNameFromHTML (==) (v) body
+  case name of 
+    Right name' -> addPostParam name' $ v
+    Left e -> failure $ e
```

Comment:
The function application operators are redundant in both cases. Also, let's try to use _slightly_ more descriptive names 🙂 

```suggestion
    Right name' -> addPostParam name' labelText
    Left err -> failure err
```